scripts/extensions/videoEditor/src/VideoEditor.tsx

```diff
@@ -0,0 +1,713 @@
+            delete body.rotate;
+        }
+        if (body.trim === `0,${this.videoRef.current!.duration}`) {
+            delete body.trim;
+        }
+        if (body.scale === 0) {
+            delete body.scale;
+        }
+        if (!isEmpty(body)) {
+            dataApi
+                .create('video_edit', {
+                    edit: body,
+                    item: this.state.article,
+                })
+                .then(() => {
+                    this.handleToggleLoading(true, gettext('Video is editing, please wait...'));
+                    this.intervalVideoEdit = window.setInterval(() => {
+                        this.props.superdesk.dataApi
+                            .findOne<IArticle>('video_edit', this.state.article._id + `?t=${Math.random()}`)
+                            .then((result) => {
+                                if (result.project?.processing?.video === false) {
+                                    clearInterval(this.intervalVideoEdit);
+                                    this.handleToggleLoading(false);
+                                    this.handleReset();
+                                    this.setState({
+                                        thumbnails: [],
+                                        videoSrc: result.project.url + `?t=${Math.random()}`,
+                                        article: {
+                                            ...this.state.article,
+                                            ...result,
+                                        },
+                                    });
+                                    this.loadTimelineThumbnails();
+                                }
+                            });
+                    }, 3000);
+                })
+                .catch((err: IErrorMessage) => {
+                    this.showErrorMessage(err);
+                    clearInterval(this.intervalVideoEdit);
+                });
+        }
+    }
+
+    checkIsDirty() {
+        if (this.videoRef.current === null || this.state.transformations.trim.end === 0) {
+            return false;
+        }
+
+        // ignore trim.end as initState don't load video duration due to videoRef can be null in componentDidMount
+        // confirm bar should not be toggled when user change crop aspect
+        if (
+            this.state.transformations.trim.end !== this.videoRef.current!.duration ||
+            !isEqual(
+                omit(this.state.transformations, ['trim.end', 'crop.aspect', 'crop.value', 'crop.scale']),
+                omit(this.initTransformations, ['trim.end', 'crop.aspect', 'crop.value', 'crop.scale']),
+            )
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    loadTimelineThumbnails() {
+        this.intervalThumbnails = window.setInterval(() => {
+            this.props.superdesk.dataApi
+                .findOne<IArticle>('video_edit', this.state.article._id + `?t=${Math.random()}`)
```

Comment:
What the purpose of getting an entire `project` here? 
`.findOne<IArticle>('video_edit', this.state.article._id + `?t=${Math.random()}`)`

If we want to load thumbnails why not to call only this:
`.findOne('video_edit', this.state.article._id + `?action=timeline&t=${Math.random()}`)` ?
If there are no thumbnails video server will run a task to create them and return 202 status, if the response status is 409, it means that the task is in the process, if it's 200 we get a list of thumbnails.